locate_fstar.mk

```diff
@@ -0,0 +1,27 @@
+# Find fstar.exe and set FSTAR_EXE to its absolute path.
+
+# If FSTAR_EXE is already externally set, we just use it. Note the use
+# of ?= everywhere below.
+
+# If FSTAR_HOME is set, we honor it and pick that F*.
+ifdef FSTAR_HOME
+FSTAR_EXE ?= $(abspath $(FSTAR_HOME)/bin/fstar.exe)
+endif
+
+# Otherwise we try to find it from the PATH.
+
+# Bash's 'type -P' is essentially 'which'. This relies on having bash
+# around, but does not require 'which'.
+FSTAR_EXE ?= $(shell bash -c 'type -P fstar.exe')
```

Comment:
maybe say why bash is preferable to which?

Reply:
Actually a good question.. I had `which` before, except that it failed on a Nix build for prosecco-green that did not have `which` in the PATH. Maybe `$(shell which fstar.exe || bash -c 'type -P fstar.exe')` would be preferable? (are you aware of any macos gotchas?)